src-self-hosted/translate_c.zig

```diff
@@ -5387,12 +5387,24 @@ fn parseCPrimaryExpr(c: *Context, it: *CTokenList.Iterator, source: []const u8,
     switch (tok.id) {
         .CharLiteral => {
             const first_tok = it.list.at(0);
-            const token = try appendToken(c, .CharLiteral, try zigifyEscapeSequences(c, source[tok.start..tok.end], source[first_tok.start..first_tok.end], source_loc));
-            const node = try c.a().create(ast.Node.CharLiteral);
-            node.* = .{
-                .token = token,
-            };
-            return &node.base;
+            if (
+                (source[tok.start+1] == '\\' and tok.end - tok.start == 4)
+                or (source[tok.start+1] != '\\' and tok.end - tok.start == 3)
```

Comment:
Character literals prefixed with `L`, `U` or `u` allow multibyte codepoints, additionally escape sequences can be longer than one byte. If we assume the C code to be valid this check can be done with `source[tok.start] != '\'' or source[tok.start + 1] == '\\' or tok.end - tok.start == 3`.